contracts/relayer/BaseRelayer.sol

```diff
@@ -0,0 +1,61 @@
+pragma solidity ^0.4.24;
+
+import "../lib/sig/ECDSA.sol";
+import "../apps/AragonApp.sol";
+import "../common/DepositableStorage.sol";
+
+
+contract BaseRelayer is AragonApp, DepositableStorage {
+    using ECDSA for bytes32;
+
+    bytes32 public constant OFF_CHAIN_RELAYER_SERVICE_ROLE = keccak256("OFF_CHAIN_RELAYER_SERVICE_ROLE");
+
+    uint256 private constant EXTERNAL_TX_COST = 21000;
+
+    string private constant ERROR_GAS_REFUND_FAIL = "RELAYER_GAS_REFUND_FAIL";
+    string private constant ERROR_NONCE_ALREADY_USED = "RELAYER_NONCE_ALREADY_USED";
+    string private constant ERROR_INVALID_SENDER_SIGNATURE = "RELAYER_INVALID_SENDER_SIGNATURE";
+
+    event FundsReceived(address indexed sender, uint256 amount);
+    event TransactionRelayed(address indexed from, address indexed to, uint256 nonce, bytes calldata);
+
+    modifier refundGas() {
+        uint256 startGas = gasleft();
+        _;
+        uint256 refund = EXTERNAL_TX_COST + startGas - gasleft();
+        require(msg.sender.send(refund), ERROR_GAS_REFUND_FAIL);
+    }
+
+    function () external payable {
+        emit FundsReceived(msg.sender, msg.value);
+    }
+
+    function initialize() public onlyInit {
+        initialized();
+        setDepositable(true);
```

Comment:
Note that by making it depositable, the call can be intercepted by the proxy depending on the gas the call has left: https://github.com/aragon/aragonOS/blob/dev/contracts/common/DepositableDelegateProxy.sol#L12-L15

To ensure this contract's fallback is called when sending ETH (besides ETH being deposited with `SELFDESTRUCT`) is by not making it depositable.